### ISSUE TITLE
False positive in rule SubstituteLongChar

### ISSUE THREAD

**Issue by jurjendijkstra** [Riverside-Software/sonar-openedge#318]
This rule is awesome, thanks! However it has a false positive in the following example:

```
SUBSTITUTE( "aaa &1 zzz", STRING(mylongcharvariable) ) .
```

mylongcharvariable is converted to a character, so the argument of the substitute function is actually not a longchar anymore. So that's the false positive.

On the bright side, I am happy with this false positive because it accidently points at the possible runtime error on the STRING function. The STRING function will raise error 9324 when mylongcharvariable is too long :-)  But that could perhaps be in a separate lint rule.

**Comment by gquerret** [comment]
Agreed it's a false positive, and that a second rule will have to be created.

**Comment by gquerret** [comment]
Second rule idea reference in #508

**Comment by gquerret** [comment]
Case in handled in VariableOverflow rule (issue #508)
